### ISSUE TITLE
remove drop implementation and requiring detach or return for promise

### ISSUE THREAD

**Issue by PolyProgrammist** [near/near-sdk-rs#1451]
https://github.com/near/near-sdk-rs/pull/1165#discussion_r2635886406

>I think the only solution really is to remove the Drop implementation and require users to explicitly .detach() (this method was recently introduced) or return the Promise, and only the returned promise will be executed (we abused serde / borsh serialization for that: https://github.com/near/near-sdk-rs/blob/master/near-sdk/src/promise.rs#L835-L859)
